### PR TITLE
Display edit link for sidebar announcements

### LINKED ISSUE #3861: Display edit link for announcements on sidebar
It has been reported that the ability to easily edit the instructions has been lost. Old project, language and TP descriptions allowed to edit in place. This probably is not a good idea, but at least some edit link must be provided for users with enough permissions to edit the announcements.

### PR DESCRIPTION
**Pull request by jleclanche**
Fixes #3861

Not sure it's right to get the `page_type` this way but couldn't figure out a better way, suggestions welcome.

### CODE REVIEW

File: pootle/templates/staticpages/_body.html [hunk 1]
```diff
@@ -9,7 +9,14 @@
   {% endblocktrans %}
   </div>
   {% endif %}
-  <h1 class="title">{{ page.title }}</h1>
+  <h1 class="title">
+    {{ page.title }}
+    {% if user.is_superuser %}
```
Comment (unho) [inline pootle/templates/staticpages/_body.html]: We want to display the edit link not only to global admins, but also to TP/Language/Project managers as for #3908.

File: pootle/templates/staticpages/_body.html [hunk 2]
```diff
@@ -9,7 +9,14 @@
...
+    <span class="edit-link">
```
Comment (unho) [inline pootle/templates/staticpages/_body.html]: Tested this on my computer, but the edit link is not displayed with smaller font. I refreshed several times and dropped cache. Perhaps it is only me, so I would like someone else to test and confirm this is displayed as expected.

I also considered this to be not in the header line since some headers can be long, but I don't think it is a major issue. In other words, this is something I can live with.

Comment (julen) [inline pootle/templates/staticpages/_body.html]: The CSS class definition is invalid.

Comment (jleclanche) [inline pootle/templates/staticpages/_body.html]: Oh that explains it, I thought I had a caching issue.

File: pootle/templates/staticpages/_body.html (hunk 1 above)
Comment (jleclanche) [inline pootle/templates/staticpages/_body.html]: How would I check for that?

File: pootle/templates/staticpages/_body.html (hunk 2 above)
Comment (unho) [inline pootle/templates/staticpages/_body.html]: Yes, missed that. It should be `.staticpage .edit-link`.

File: pootle/templates/staticpages/_body.html (hunk 1 above)
Comment (unho) [inline pootle/templates/staticpages/_body.html]: The following allows to check if a specific user can administrate a TP:

```python
check_user_permission(user, 'administrate', tp.directory)
```

I cannot find any specific permissions for administrating a project or a language, so perhaps you should use the `change_project` and `change_language` permissions.

Comment (unho) [inline pootle/templates/staticpages/_body.html]: Or perhaps use `view` permission, but that doesn't seem related to management. @julen Any idea?

Comment (jleclanche) [inline pootle/templates/staticpages/_body.html]: Which types of users are allowed to edit this? Ideally I'd like to show the link to anyone who can edit it.

Comment (unho) [inline pootle/templates/staticpages/_body.html]: By type of announcement:

- Project announcement: Can be edited by global admins and project managers
- Language announcement: Can be edited by global admins and language managers
- TP announcement: Can be edited by global admins and TP managers

As commented above we don't seem to have any permission to tell if a user is a project manager or language manager.

Comment (unho) [inline pootle/templates/staticpages/_body.html]: Just noticed that perhaps the admin UI for editing announcements must be also adjusted so the displayed announcements are restricted to those having only the required permissions.

Comment (unho) [inline pootle/templates/staticpages/_body.html]: After discussion on #3908, and given that the announcements are only displayed on TPs, it is ok to display the edit link if the user is a superuser or has `administrate` permission in the current TP.

Comment (unho) [inline pootle/templates/staticpages/_body.html]: Noticed that this is a template that is included both in the template that displays staticpages (including announcements) and in the browser template to render announcements in the sidebar. In the former we probably can't check for TP `administrate` permission since it is completely detached from any particular TP.

File: pootle/templates/staticpages/_body.html (hunk 2 above)
Comment (unho) [inline pootle/templates/staticpages/_body.html]: For the record this particular issue is now fixed.

File: pootle/templates/staticpages/_body.html [hunk 3]
```diff
@@ -9,7 +9,14 @@
...
+      [<a href="{% url 'pootle-staticpages-edit' pk=page.id page_type='announcements' %}">edit</a>]
```
Comment (unho) [inline pootle/templates/staticpages/_body.html]: The following concern was raised on the PR initial message:

> Not sure it's right to get the `page_type` this way but couldn't figure out a better way, suggestions welcome.

I don't like the idea of hardcoding this. I would suggest to create a new URL on `staticpages` app named `pootle-staticpages-announcement-edit` that imports `ANN_TYPE` from `staticpages.views` and uses it for the URL pattern. This way it won't be necessary to hardcode anything on the template.

@julen Thoughts on this?

Comment (unho) [inline pootle/templates/staticpages/_body.html]: Forget about this. Noticed that `page.get_edit_url` should provide us the URL without need to use the `url` templatetag.

**Comment by unho** [comment]
Merged in 87e0474a with some minor fixes.

Note that finally was decided not to display the link if the user has `administrate` permission since doing that would require to alter the admin for staticpages so users with such permission can edit specific announcements. Also it would require altering the view to see the staticpages (like http://127.0.0.1:8000/pages/announcements/projects/firefox/) in order to check for the `administrate` permission as well.

File: pootle/templates/staticpages/_body.html (hunk 3 above)
Comment (julen) [inline pootle/templates/staticpages/_body.html]: This lacks i18n of any sort.

Comment (unho) [inline pootle/templates/staticpages/_body.html]: Missed that. Landed a fix in e7db2b13.
